postgres/coredb-pg-base/Dockerfile

```diff
@@ -1,71 +1,119 @@
-	mkdir -p /var/lib/postgresql; \
-	chown -R postgres:postgres /var/lib/postgresql
-
-RUN apt-get update && apt-get install -y \
-    curl \
-    ca-certificates \
-    gnupg \
-    lsb-release \
-    libreadline-dev \
-    zlib1g-dev \
-    libpq-dev \
-    && rm -rf /var/lib/apt/lists/*
+      groupadd -r postgres --gid=999; \
+      useradd -r -g postgres --uid=999 --home-dir=/var/lib/postgresql --shell=/bin/bash postgres; \
+      mkdir -p /var/lib/postgresql; \
+      chown -R postgres:postgres /var/lib/postgresql; \
+      apt-get update; apt-get install -y curl ca-certificates gnupg lsb-release lbzip2
 
 STOPSIGNAL SIGINT
 
-ENV PGDATA /var/lib/postgresql/data
-ENV PG_VERSION 15.3
-ENV PG_MAJOR 15
-ENV PATH $PATH:/usr/lib/postgresql/$PG_MAJOR/bin
-
 RUN set -eux; \
-	apt-get update; apt-get install -y --no-install-recommends locales; rm -rf /var/lib/apt/lists/*; \
-	localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8
+      apt-get update; \
+      apt-get install -y --no-install-recommends \
+      locales \
+    ; \
+    rm -rf /var/lib/apt/lists/*; \
+	  localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8
 ENV LANG en_US.utf8
 
 RUN mkdir /docker-entrypoint-initdb.d
 
-# Install postgres and some extensions
-RUN apt-get update && apt-get install -y \
-        build-essential \
-        zlib1g-dev \
-        libreadline-dev \
-        libssl-dev \
-        && rm -rf /var/lib/apt/lists/*
-
 # Build Postgres from source
-RUN curl https://ftp.postgresql.org/pub/source/v15.3/postgresql-15.3.tar.bz2 -o postgresql-15.3.tar.bz2
+RUN curl https://ftp.postgresql.org/pub/source/v${PG_VERSION}/postgresql-${PG_VERSION}.tar.bz2 -o postgresql-${PG_VERSION}.tar.bz2
 RUN tar xf postgresql-${PG_VERSION}.tar.bz2
+RUN set -eux; \
+      apt-get update && apt-get install -y \
+          libreadline-dev \
+          zlib1g-dev \
+          libpq-dev \
+          build-essential \
+          python3-dev \
+          tcl-dev \
+          libxslt1-dev \
+          libperl-dev \
+          libpam0g-dev \
+          libreadline-dev \
+          libssl-dev \
+          xz-utils \
+          libnss-wrapper \
+          llvm \
+          clang \
+          icu-devtools \
+          pkg-config \
+          libgss-dev \
+          libkrb5-dev \
+          uuid-dev \
+          gettext \
+          liblz4-dev \
+          libsystemd-dev \
+          libselinux1-dev \
+          libzstd-dev \
+      ; \
+      apt-get autoremove -y; \
+      apt-get clean -y; \
+      rm -rf /var/lib/apt/lists/*
 WORKDIR postgresql-${PG_VERSION}
-RUN ./configure --prefix=/usr/lib/postgresql/${PG_MAJOR} --datarootdir=/usr/share --with-openssl
-RUN make
+ENV CFLAGS "-g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer"
+ENV LDFLAGS "-Wl,-z,relro -Wl,-z,now" 
+RUN ./configure --prefix=/usr/lib/postgresql/${PG_MAJOR} \
+        --datarootdir=${ALTDIR} \
+        --libdir=${ALTDIR}/${PG_MAJOR}/lib \
```

Comment:
I noticed that we have lib dir as subdirectory of share dir:

```
Status: Downloaded newer image for quay.io/coredb/coredb-pg-base:f3a6fdd
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
postgres@d2bfa44cc1ea:/postgresql-15.3$ pg_config --sharedir
/var/lib/postgresql/data/tembo
postgres@d2bfa44cc1ea:/postgresql-15.3$ ls /var/lib/postgresql/data/
tembo
postgres@d2bfa44cc1ea:/postgresql-15.3$ pg_config --pkglibdir
/var/lib/postgresql/data/tembo/15/lib
```

In the current situation, they are not subdirectories of each other:
```
Status: Downloaded newer image for quay.io/coredb/coredb-pg-slim:latest
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
postgres@aa60a49361f7:/$ pg_config --sharedir
/usr/share/postgresql/15
postgres@aa60a49361f7:/$ pg_config --pkglibdir
/usr/lib/postgresql/15/lib
```

What do you think about that?

Reply:
Yeah for now they are under the same directory.  We can move them later, but this allows a single copy back into `$PGDATA` instead of two.